mllib/src/test/scala/org/apache/spark/ml/optim/WeightedLeastSquaresSuite.scala

```diff
@@ -74,6 +89,35 @@ class WeightedLeastSquaresSuite extends SparkFunSuite with MLlibTestSparkContext
     }
   }
 
+  test("WLS against lm when label is constant") {
+    /*
+       R code:
+       # here b is constant
+       df <- as.data.frame(cbind(A, b))
+       for (formula in c(b ~ . -1, b ~ .)) {
+         model <- lm(formula, data=df, weights=w)
+         print(as.vector(coef(model)))
+       }
+
+      [1] -9.221298  3.394343
+      [1] 17  0  0
+    */
+
+    val expected = Seq(
+      Vectors.dense(0.0, -9.221298, 3.394343),
+      Vectors.dense(17.0, 0.0, 0.0))
+
+    var idx = 0
+    for (fitIntercept <- Seq(false, true)) {
+      val wls = new WeightedLeastSquares(
+        fitIntercept, regParam = 0.0, standardizeFeatures = false, standardizeLabel = true)
+        .fit(instancesConstLabel)
```

Comment:
I've added exception and the test for `standardizeLabel = true`, `regParam != 0` and `yStd == 0.0`. The only thing now left is to add test for the case `standardizeLabel = false`, `regParam != 0` and `yStd == 0.0`. As I mentioned before, in this case I cannot compare with `glmnet`. So, I'll try to implement normal equation in python by myself and compare with that. The good thing is that, for this  particular case, both `normal` and `l-bfgs` give identical results!